src/Tasks/AssemblyDependency/TaskItemSpecFilenameComparer.cs

```diff
@@ -47,21 +47,38 @@ public int Compare(object o1, object o2)
 
         public int Compare(ITaskItem x, ITaskItem y)
         {
-            if (Object.ReferenceEquals(x, y))
+            if (ReferenceEquals(x, y))
             {
                 return 0;
             }
 
-            string f1 = Path.GetFileName(x.ItemSpec);
-            string f2 = Path.GetFileName(y.ItemSpec);
+            string xItemSpec = x.ItemSpec;
+            string yItemSpec = y.ItemSpec;
+
+            int xFilenameStart = xItemSpec.LastIndexOfAny(FileMatcher.directorySeparatorCharacters);
+            if (xFilenameStart == -1)
+            {
+                xFilenameStart = 0;
+            }
+
+            int yFilenameStart = yItemSpec.LastIndexOfAny(FileMatcher.directorySeparatorCharacters);
+            if (yFilenameStart == -1)
+            {
+                yFilenameStart = 0;
+            }
 
-            int fileComparison = String.Compare(f1, f2, StringComparison.OrdinalIgnoreCase);
+            int fileComparison = string.Compare(xItemSpec,
+                xFilenameStart,
+                yItemSpec,
+                yFilenameStart,
+                int.MaxValue, // all characters after the start index
```

Comment:
I just learned!

1. Assume the API is just `string1, start1, string2, start2, comparisonType`
2. Get kinda baffled by the red squiggle.
2. Oh, it requires a length.
3. Write code to get the length of one side and use that.
4. Get annoyed at having to do that.
5. Look at the implementation and discover that it already does match-portion length extraction and [takes a minimum](https://github.com/dotnet/coreclr/blob/720800cb23024f36c3f281fb586107e028b14b75/src/mscorlib/src/System/String.Comparison.cs#L612-L613) with that and this param.
5. Rejoice!